apps/src/templates/projects/PersonalProjectsTable.jsx

```diff
@@ -0,0 +1,298 @@
+import React, {PropTypes} from 'react';
+import FontAwesome from '@cdo/apps/templates/FontAwesome';
+import i18n from "@cdo/locale";
+import color from "../../util/color";
+import {ImageWithStatus} from '../ImageWithStatus';
+import {Table, sort} from 'reactabular';
+import wrappedSortable from '../tables/wrapped_sortable';
+import orderBy from 'lodash/orderBy';
+import {PROJECT_TYPE_MAP} from './projectConstants';
+
+const PROJECT_DEFAULT_IMAGE = '/blockly/media/projects/project_default.png';
+
+const THUMBNAIL_SIZE = 65;
+
+/** @enum {number} */
+export const COLUMNS = {
+  THUMBNAIL: 0,
+  PROJECT_NAME: 1,
+  APP_TYPE: 2,
+  LAST_EDITED: 3,
+  PUBLIC_GALLERY: 4,
+  ACTIONS: 5,
+};
+
+const styles = {
+  table: {
+    width: 970,
+    borderRadius: 5,
+    color: color.charcoal,
+    backgroundColor: color.table_light_row
+  },
+  cell: {
+    border: '1px solid',
+    borderColor: color.border_light_gray,
+    padding: 10,
+    fontSize: 14,
+  },
+  headerCell: {
+    border: '1px solid',
+    borderColor: color.border_light_gray,
+    padding: 20,
+    backgroundColor: color.lightest_gray,
+    color: color.charcoal
+  },
+  cellThumbnail: {
+    borderWidth: '1px 0px 1px 1px',
+    borderColor: color.border_light_gray,
+    width: THUMBNAIL_SIZE,
+    minWidth: THUMBNAIL_SIZE,
+    padding: 2
+  },
+  headerCellThumbnail: {
+    borderWidth: '1px 0px 1px 1px',
+    borderColor: color.border_light_gray,
+    padding: 0
+  },
+  cellName: {
+    borderWidth: '1px 1px 1px 0px',
+    borderColor: color.border_light_gray,
+    padding: 15,
+    width: 270
+  },
+  headerCellName: {
+    borderWidth: '1px 1px 1px 0px',
+    borderColor: color.border_light_gray,
+    padding: 15
+  },
+  cellType: {
+    width: 120
+  },
+  cellIsPublished: {
+    textAlign: 'center'
+  },
+  cellAction: {
+    textAlign: 'center'
+  },
+  thumbnailWrapper: {
+    height: THUMBNAIL_SIZE,
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  link: {
+    color: color.teal,
+    fontFamily: '"Gotham 5r", sans-serif',
+    fontSize: 14,
+    textDecoration: 'none'
+  }
+};
+
+// Table of user's projects
+// TODO(caleybrock): add an actions component to allow users to modify projects
+const PersonalProjectsTable = React.createClass({
+  propTypes: {
+    projectList: PropTypes.array.isRequired
```

Comment:
should we specify the shape of the objects within the array (i.e. using arrayOf)